vm_supervisor/storage.py

```diff
@@ -155,28 +161,93 @@ def create_ext4(path: Path, size_mib: int) -> bool:
     return True
 
 
+async def create_devmapper(volume: PersistentVolume | RootfsVolume, namespace: str) -> Path:
+    volume_name = volume.name if isinstance(volume, PersistentVolume) else "rootfs"
+    volume_dev_name = f"{namespace}_{volume_name}"
+    path_volume_dev_name = Path(f"/dev/mapper/{volume_dev_name}")
+    if path_volume_dev_name.is_block_device():
+        return path_volume_dev_name
+    path = Path(
+        join(settings.PERSISTENT_VOLUMES_DIR, namespace, f"{volume_name}.ext4")
+    )
+    if not os.path.isfile(path):
+        logger.debug(f"Creating {volume.size_mib}MB volume")
+        os.system(f"dd if=/dev/zero of={path} bs=1M count={volume.size_mib}")
+        if settings.USE_JAILER:
+            os.system(f"chown jailman:jailman {path}")
+    try:
+        parent_path = await get_runtime_path(volume.parent.ref)
+        loop_base = subprocess.run(
+            ["losetup", "--find", "--show", "--read-only", parent_path],
+            check=True,
+            capture_output=True,
+            encoding="UTF-8").stdout.strip()
+        root_size = subprocess.run(
+            ["blockdev", "--getsz", parent_path],
+            check=True,
+            capture_output=True,
+            encoding="UTF-8").stdout.strip()
+        volume_data_size = subprocess.run(
+            ["blockdev", "--getsz", path],
+            check=True,
+            capture_output=True,
+            encoding="UTF-8").stdout.strip()
+        loop_user_data = subprocess.run(
+            ["losetup", "--find", "--show", path],
+            check=True,
+            capture_output=True,
+            encoding="UTF-8").stdout.strip()
+        table_command = f"0 {root_size} linear {loop_base} 0\n{root_size} {volume_data_size} zero"
+        base_dev_name = f"{namespace}_{volume_name}_base"
+        os.system(f" printf \"{table_command}\" | dmsetup create {base_dev_name}")
+        path_base_dev_name = f"/dev/mapper/{base_dev_name}"
+        table_command = f"0 {volume_data_size} snapshot {path_base_dev_name} {loop_user_data} P 8"
```

Comment:
How did you determine the chunk size (8)?

Reply:
I don't know exactly, I set it using [this](https://community.aleph.im/t/deploying-mutable-vm-instances-on-aleph/56/2) example command, we can investigate how we can ensure that is the correct chunk size.